File: src/main/java/com/ascargon/rocketshow/health/HealthStatus.java
package com.ascargon.rocketshow.health;

import com.ascargon.rocketshow.util.DeviceInformation;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
public class HealthStatus {

    private HealthStatusSeverity healthStatusSeverity = HealthStatusSeverity.OK;

    private DeviceInformation deviceInformation;
    private Long freeDiskSpacePercentage;
    private Long freeMemory;
    private Double temperature;
    private int recentErrorRate;
    private String softwareVersion;
    private Date softwareDate;
    private String raucSlot;

    private List<String> reasons = new ArrayList<>();

    public String toFailureString() {
        StringBuilder sb = new StringBuilder("Health failure: ");

        if (reasons != null && !reasons.isEmpty()) {
            sb.append(" reasons=").append(reasons);
        }

        return sb.toString();
    }

}

File: src/main/webapp/src/app/models/health-status.ts
import { DeviceInformation } from "./device-information";

export class HealthStatus {
  healthStatusSeverity: string;
  deviceInformation: DeviceInformation;
  freeDiskSpacePercentage: number;
  freeMemory: number;
  temperature: number;
  recentErrorRate: number;
  softwareVersion: string;
  softwareDate: string;
  eepromVersionDate: string;
  raucSlot: string;
  reasons: string[] = [];

  constructor(data?: any) {
    if (!data) {
      return;
    }

    this.healthStatusSeverity = data.healthStatusSeverity;
    this.deviceInformation = data.deviceInformation
      ? new DeviceInformation(data.deviceInformation)
      : undefined;
    this.freeDiskSpacePercentage = data.freeDiskSpacePercentage;
    this.freeMemory = data.freeMemory;
    this.temperature = data.temperature;
    this.recentErrorRate = data.recentErrorRate;
    this.softwareVersion = data.softwareVersion;
    this.softwareDate = data.softwareDate;
    this.eepromVersionDate = data.eepromVersionDate;
    this.raucSlot = data.raucSlot;
    this.reasons = data.reasons || [];
  }
}
